src/figuresClasses.ts

```diff
@@ -1,19 +1,85 @@
-export interface Figure {
-
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
 }
 
-export class Triangle {
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
+}
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Circle {
+export class Triangle implements Figure {
+  shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const condition = [this.a, this.b, this.c];
```

Comment:
```suggestion
    const conditions = [this.a, this.b, this.c];
```

Reply:
No, pack here **all** conditions and then check it with the `some`

Reply:
guess understood.